sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -1155,20 +1219,174 @@ public virtual Response SetAccessPolicy(IEnumerable<SignedIdentifier> tableAcl,
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns><see cref="Response{T}"/> containing a <see cref="TableBatchResponse"/>.</returns>
+        /// <exception cref="RequestFailedException"/> if the batch transaction fails./>
+        /// <exception cref="InvalidOperationException"/> if the batch has been previously submitted.
+        public virtual async Task<Response<TableBatchResponse>> SubmitTransactionAsync(
+            TableTransactionalBatch transactionalBatch,
+            CancellationToken cancellationToken = default) =>
+            await SubmitBatchAsyncInternal(transactionalBatch, true, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Submits the batch transaction to the service for execution.
+        /// The sub-operations contained in the batch will either succeed or fail together as a transaction.
+        /// </summary>
+        /// <param name="transactionalBatch">The <see cref="TableTransactionalBatch"/> to submit to the service.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns><see cref="Response{T}"/> containing a <see cref="TableBatchResponse"/>.</returns>
+        /// <exception cref="RequestFailedException"/> if the batch transaction fails./>
+        /// <exception cref="InvalidOperationException"/> if the batch has been previously submitted.
+        public virtual Response<TableBatchResponse> SubmitTransaction(TableTransactionalBatch transactionalBatch, CancellationToken cancellationToken = default) =>
+            SubmitBatchAsyncInternal(transactionalBatch, false, cancellationToken).EnsureCompleted();
+
+        internal async Task<Response<TableBatchResponse>> SubmitBatchAsyncInternal(
+            TableTransactionalBatch transactionalBatch,
+            bool async,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(transactionalBatch, nameof(transactionalBatch));
+
+            using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TableClient)}.{nameof(SubmitTransaction)}");
+            scope.Start();
+            try
+            {
+                var _batchOperations = new TableRestClient(_diagnostics, CreateBatchPipeline(), _tableOperations.endpoint, _tableOperations.clientVersion);
+                if (transactionalBatch._requestMessages.IsEmpty)
+                {
+                    throw new InvalidOperationException(TableConstants.ExceptionMessages.BatchIsEmpty);
+                }
+                if (transactionalBatch._submitted)
+                {
+                    throw new InvalidOperationException(TableConstants.ExceptionMessages.BatchCanOnlyBeSubmittedOnce);
+                }
+                transactionalBatch._submitted = true;
```

Comment:
If this operation fails, doesn't this prevent retries?   Should we consider unsetting in the `catch` block, or is the intent to create a new batch and use that?